src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php

```diff
@@ -0,0 +1,311 @@
+    {
+        $tokensAnalyzer = new TokensAnalyzer($tokens);
+
+        for ($i = $endIndex - 1; $i > $startIndex; --$i) {
+            if (!$tokens[$i]->isGivenKind(T_FUNCTION) || $tokensAnalyzer->isLambda($i)) {
+                continue;
+            }
+
+            $functionIndex = $i;
+            $docBlockIndex = $i;
+
+            // ignore abstract functions
+            $braceIndex = $tokens->getNextTokenOfKind($functionIndex, [';', '{']);
+            if (!$tokens[$braceIndex]->equals('{')) {
+                continue;
+            }
+
+            do {
+                $docBlockIndex = $tokens->getPrevNonWhitespace($docBlockIndex);
+            } while ($tokens[$docBlockIndex]->isGivenKind([T_PUBLIC, T_PROTECTED, T_PRIVATE, T_FINAL, T_ABSTRACT, T_COMMENT]));
+
+            if (!$tokens[$docBlockIndex]->isGivenKind(T_DOC_COMMENT)) {
+                continue;
+            }
+
+            $doc = new DocBlock($tokens[$docBlockIndex]->getContent());
+            $annotations = [];
+
+            foreach ($doc->getAnnotationsOfType([
+                'expectedException',
+                'expectedExceptionCode',
+                'expectedExceptionMessage',
+                'expectedExceptionMessageRegExp',
+            ]) as $annotation) {
+                $tag = $annotation->getTag()->getName();
+                $content = $this->extractContentFromAnnotation($annotation);
+                $annotations[$tag] = $content;
+                $annotation->remove();
+            }
+
+            if (!isset($annotations['expectedException'])) {
+                continue;
+            }
+            if (!$this->fixMessageRegExp && isset($annotations['expectedExceptionMessageRegExp'])) {
+                continue;
+            }
+
+            $originalIndent = $this->detectIndent($tokens, $docBlockIndex);
+
+            $paramList = $this->annotationsToParamList($annotations);
+
+            $newMethodsCode = '<?php $this->'
+                .(isset($annotations['expectedExceptionMessageRegExp']) ? 'setExpectedExceptionRegExp' : 'setExpectedException')
+                .'('
+                .implode($paramList, ', ')
+                .');';
+            $newMethods = Tokens::fromCode($newMethodsCode);
+            $newMethods[0] = new Token([
+                T_WHITESPACE,
+                $this->whitespacesConfig->getLineEnding().$originalIndent.$this->whitespacesConfig->getIndent(),
+            ]);
+
+            // apply changes
+            $tokens[$docBlockIndex] = new Token([T_DOC_COMMENT, $doc->getContent()]);
+            $tokens->insertAt($braceIndex + 1, $newMethods);
+
+            $i = $docBlockIndex;
+        }
+    }
+
+    /**
+     * @param Annotation $annotation
+     *
+     * @return string
+     */
+    private function extractContentFromAnnotation(Annotation $annotation)
+    {
+        $tag = $annotation->getTag()->getName();
+
+        preg_match('/^\s*\*\s*@'.$tag.'\s+(.+)$/', $annotation->getContent(), $matches);
```

Comment:
this would match ` * @test test @internal` with the `@internal` tag in it, maybe best to stop on the next `@`?

Reply:
`* @test test @internal`
if it's one line, it's one tag, and content is `test @internal`

I added test to prove it's taken as content and put to `->expectExceptionMessage('test @internal')`